torch_np/_funcs_impl.py

```diff
@@ -909,6 +909,20 @@ def take_along_axis(arr: ArrayLike, indices: ArrayLike, axis):
     return torch.take_along_dim(arr, indices, axis)
 
 
+def put(
+    a: ArrayLike,
+    ind: ArrayLike,
+    v: ArrayLike,
+    mode: NotImplementedType = "raise",
+):
+    if v.numel() < ind.numel():
+        numel_ratio = float(ind.numel() / v.numel())
+        if numel_ratio.is_integer():
+            v = torch.stack([v for _ in range(int(numel_ratio))])
```

Comment:
This logic looks a bit odd. 

So, you do this stacking (which should probably be done via `torch.repeat`) when the number of elements of v divides that of ind, and otherwise we do nothing (and I guess we error out inside `a.put_`?

Do you have a reference for where this is implemented in NumPy?

Reply:
ISTM it's this one: https://github.com/numpy/numpy/blob/main/numpy/core/src/multiarray/item_selection.c#L350

Reply:
So it seems like it also wraps when they are not divisible?
https://github.com/numpy/numpy/blob/6073588dd73809a60819d71b9527194195f73f08/numpy/core/src/multiarray/item_selection.c#L501
See also
```python
>>> import numpy as np
>>> x = np.array([1,2,3,4])
>>> x.put([-1, -2, -3], [-8, -7])
>>> x
array([ 1, -8, -7, -8])
```

Reply:
I don't think we have a function that does "repeat until you fit this shape", but you can use `torch.repeat` and then take as many elements from it as you need via advanced indexing or `index_select`. This second part can be guarded by a divisibility test, so that we just do it when it's strictly necessary.

Reply:
For some reason thought `int / int` could return both ints and float-ints, hence the unnecessary `float()` casting. Removed!

> you can use `torch.repeat` and then take as many elements from it as you need via advanced indexing or `index_select`

I don't quite get this sorry—as I read it, you'd need to workout a repeat which is the same-size/bigger-than the indexer anywho. Pushed a change, lemme know what you think.

<br />
<sup>
<blockquote>Do you have a reference for where this is implemented in NumPy?</blockquote>

Haven't been checking the NumPy implementations for these things, should start I suppose :sweat_smile:
</sup>